packages/server/src/whip/whipEndpoint.ts

```diff
@@ -1,150 +1,181 @@
+    }    
+    this.server = fastify({ 
+      ignoreTrailingSlash: true, 
+      logger: { level: "info" },
+      https: httpsServer,
+    });
+    this.server.register(require("fastify-cors"), {
+      exposedHeaders: ["Location", "ETag", "Link", "Access-Control-Allow-Methods"],
+      methods: ["POST", "GET", "OPTIONS", "DELETE", "PATCH"],
+      preflightContinue: true,
+      strictPreflight: false,
+    });
+    this.server.register(api, { prefix: "/api/v2", instance: this });
+    this.server.get("/", async () => {
+      return "OK\n";
+    });
+    this.resources = {};
+  }
+
+  registerBroadcasterClient(broadcasterClientSfuPair: BroadcasterClientSfuPair) {
+    this.broadcasterClientSfuPairs.push(broadcasterClientSfuPair);
+  }
+
+  setOriginSfuUrl(url: string) {
+    this.originSfuUrl = url;
+  }
+
+  setSfuApiKey(apiKey: string) {
+    this.sfuApiKey = apiKey;
+  }
+
+  hasBroadcasterClient(): boolean {
+    return this.broadcasterClientSfuPairs.length !== 0;
+  }
+
+  getBroadcasterClientSfuPairs(): BroadcasterClientSfuPair[] {
+    return this.broadcasterClientSfuPairs;
+  }
+
+  getOriginSfuUrl(): string {
+    return this.originSfuUrl;
+  }
+
+  getSfuApiKey(): string {
+    return this.sfuApiKey;
+  }
+
+  addResource(resource: WhipResource) {
+    this.resources[resource.getId()] = resource;
+  }
+
+  deleteResource(id: string) {
+    const resource = this.resources[id]; 
+    if (resource) {
+      delete this.resources[id];
+      resource.destroy();
+    }
+  }
+
+  async patchResource(id: string, body: string, eTag:string | undefined): Promise<number> {
+    const resource = this.resources[id]; 
+    if (resource) {
+      return resource.patch(body, eTag);
+    }
+    return 404;
+  }
+
+  listResources() {
+    return Object.keys(this.resources).map(id => {
+      const resource = this.resources[id];
+      return { id: resource.getId(), type: resource.getType() };
+    });
+  }
+
+  getEnabledPlugins() {
+    return this.enabledWrtcPlugins;
+  }
+
+  getResourceById(resourceId: string) {
+    return this.resources[resourceId];
+  }
+
+  getIceServers(): WhipResourceIceServer[]|null {
+    return this.iceServers;
+  }
+
+  getServerAddress(): string | undefined {
+    if (this.hostname) {
+      return (this.useHttps ? "https" : "http") + "://" + this.hostname + ":" + this.extPort;
+    }
+    return undefined;
+  }
+
+  readEdgeListFromFile(): OriginsAndEdgesPerTerritory {
+    const sfuConfigFileContents = readFileSync(this.jsonFilePath);
+    return <OriginsAndEdgesPerTerritory>JSON.parse(sfuConfigFileContents.toString());
+  }
+
+  async readEdgeListFromService(): Promise<OriginsAndEdgesPerTerritory> {
+    return (await callResourceManager(process.env.RESOURCE_MANAGER_URL)) as OriginsAndEdgesPerTerritory;
+  }
+
+  useResourceManager(): boolean {
+    if(this.resourceManagerUrl) {
+      console.log(`Env RESOURCE_MANAGER_URL is set. Getting sfu-config from Resource Manager.`);

```

Comment:
This is not exactly true as you would set the resourceManagerUrl via constructor options and not the environment variable. Remember that the WHIP Endpoint is a library and the `server.ts` is just an example / reference implementation of how to use it. Anything that is specific to the server.ts implementation should not be in the library.